### PR TITLE
CU/7720-Dylan-Last10Dependencies

### PR DESCRIPTION
**Pull request by Sparowhawk**
## Description of Change
updated the following dependencies:

"@tanstack/react-query": "^5.17.19",
"jsc-android": "^250231.0.0",
"ts-node": "^10.9.2",
"@types/node": "^20.11.5",
"concurrently": "^8.2.2",
"jsc-android": "^250231.0.0",
"i18next": "^23.7.18",
"react-i18next": "^14.0.1",

Prettier resolution is being updated with @alexandec update to the linting rules

“[@types/styled-components-react-native](https://github.com/types/styled-components-react-native)”: “^5.2.1",
“styled-components”: “^5.3.10”,
I am letting the design-system team update this dependency by moving components over to the design library and eventually updating this. When I tried to update these 2 locally there were lots of issues that would require me to rewrite the components which I didn't want to do.

    “lottie-react-native”: “^5.1.2”, is also not being updated because the design team will be moving the loading component over in April

## Screenshots/Video
N/A

## Testing
yarn test/sniff tests

- [x] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations
“react-i18next”: “^11.18.6”,
“i18next”: “^21.6.10”,

“@tanstack/react-query”: “^4.33.0”,
“lottie-react-native”: “^5.1.2”,
“jsc-android”: “^250230.2.1”,
“@types/node”: “^18.11.9",
“@types/styled-components-react-native”: “^5.2.1",
“styled-components”: “^5.3.10”,
“concurrently”: “^7.0.0",
“ts-node”: “^10.3.1",

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

File: VAMobile/src/testUtils.tsx [hunk 1]
```diff
@@ -204,12 +204,6 @@ function render(ui, { preloadedState, navigationProvided = false, queriesData, .
           retry: false,
         },
       },
-      logger: {
-        log: console.log,
-        warn: console.warn,
-        // Silence the error console
-        error: () => {},
-      },
```
Comment (Sparowhawk) [inline VAMobile/src/testUtils.tsx:212]: This is deprecated

"Custom loggers were already deprecated in 4 and have been removed in this version. Logging only had an effect in development mode, where passing a custom logger is not necessary."

**Comment by Sparowhawk** [comment]
react query isLoading is also updated to isPending for mutations

File: VAMobile/src/components/Carousel/CarouselTabBar.tsx [hunk 2]
```diff
@@ -78,7 +78,7 @@ function CarouselTabBar({ onCarouselEnd, screenList, translation }: CarouselTabB
   }
 
   const goBackOrSkipBtn = () => {
-    let onPressCallback: TFunction
+    let onPressCallback: () => void
```
Comment (Sparowhawk) [inline VAMobile/src/components/Carousel/CarouselTabBar.tsx:81]: no idea why this was listed as TFunction to begin with lol

**Review by alexandec — APPROVED** [review]

**Review by theodur — APPROVED** [review]
Changes look good ✅  I went through the [React Query migration guide](https://tanstack.com/query/latest/docs/react/guides/migrating-to-v5) and verified we didn't miss anything

**Review by rbontrager — APPROVED** [review]

**Pull request merged**